plugins/inputs/mavlink/mavlink_test.go

```diff
@@ -0,0 +1,268 @@
+					TimeBootMs: 123,
+					Roll:       1.234,
+					Pitch:      0.463,
+					Yaw:        -0.112,
+					Rollspeed:  0.001,
+					Pitchspeed: 0.002,
+					Yawspeed:   0.003,
+				},
+				Checksum: 0,
+			},
+			expected: []telegraf.Metric{metric.New(
+				"attitude",
+				map[string]string{
+					"sys_id": "1",
+				},
+				map[string]interface{}{
+					"pitch":        float64(0.463),
+					"roll":         float64(1.234),
+					"yaw":          float64(-0.112),
+					"pitchspeed":   float64(0.002),
+					"rollspeed":    float64(0.001),
+					"yawspeed":     float64(0.003),
+					"time_boot_ms": uint64(123),
+				},
+				time.Unix(0, 0),
+			)},
+		},
+		{
+			name: "ESC Status",
+			input: &frame.V2Frame{
+				SequenceNumber: 1,
+				SystemID:       1,
+				ComponentID:    1,
+				Message: &common.MessageEscStatus{
+					Index:    0,
+					TimeUsec: 12345,
+					Rpm:      [4]int32{0, 1, 2, 3},
+					Voltage:  [4]float32{10.0, 11.0, 12.0, 13.0},
+					Current:  [4]float32{14.0, 15.0, 16.0, 17.0},
+				},
+				Checksum: 0,
+			},
+			expected: []telegraf.Metric{metric.New(
+				"esc_status",
+				map[string]string{
+					"sys_id": "1",
+				},
+				map[string]interface{}{
+					"index":     uint64(0),
+					"time_usec": uint64(12345),
+					"current_1": float64(14.0),
+					"current_2": float64(15.0),
+					"current_3": float64(16.0),
+					"current_4": float64(17.0),
+					"rpm_1":     int32(0),
+					"rpm_2":     int32(1),
+					"rpm_3":     int32(2),
+					"rpm_4":     int32(3),
+					"voltage_1": float64(10.0),
+					"voltage_2": float64(11.0),
+					"voltage_3": float64(12.0),
+					"voltage_4": float64(13.0),
+				},
+				time.Unix(0, 0),
+			)},
+		},
+	}
+
+	tmpFilter, err := filter.Compile(make([]string, 0))
+	require.NoError(t, err)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actual := convertFrameToMetric(tt.input, tmpFilter)
+			testutil.RequireMetricsStructureEqual(t, tt.expected, []telegraf.Metric{actual}, testutil.IgnoreTime())
+		})
+	}
+}
+
+// Test that the plugin can interface with a real ArduPilot container.
+func TestArduPilotIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	// Start the docker container for ArduPilot
+	t.Log("Starting ArduPilot container")
+	container := testutil.Container{
+		Image:        "radarku/ardupilot-sitl",
+		ExposedPorts: []string{"5760"},
+		WaitingFor: wait.ForAll(
+			wait.ForListeningPort(nat.Port("5760")),
+		),
+	}
+	require.NoError(t, container.Start(), "failed to start ardupilot container")
+	defer container.Terminate()
+
+	t.Logf("ArduPilot is listening on port: %s", container.Ports["5760"])
+
+	// Setup the plugin
+	plugin := Mavlink{
+		URL:      "tcp://127.0.0.1:" + container.Ports["5760"],
+		SystemID: 254,
+	}
+	plugin.Log = testutil.Logger{}
+	require.NoError(t, plugin.Init())
+
+	// Collect the metrics and compare
+	var acc testutil.Accumulator
+	require.NoError(t, plugin.Start(&acc))
```

Comment:
We need to make sure to stop the plugin even if the tests fail
```suggestion
	require.NoError(t, plugin.Start(&acc))
	defer plugin.Stop()
```